### ISSUE TITLE
Rank "n" evaluation of fisher information in jupyter notebook

### ISSUE THREAD

**Issue by simomagi** [DLR-RM/curvature#5]
In the jupyter notebook, the rank-10 diagonal FiM approximation is computed as follows:
`    

    # A rank-10 diagonal FiM approximation.
    for sample in range(10):
        labels = dist.sample()
        
        loss = criterion(logits, labels)
        model.zero_grad()
        loss.backward(retain_graph=True)

        diag.update(batch_size=images.size(0))
`
Since we are performing a montecarlo sampling of the output distribution is it possible that we should divide by 1/ sqrt(n_trials) the loss function before we compute loss.backward() ? I.E. 
    
    # A rank-10 diagonal FiM approximation.
    for sample in range(10):
        labels = dist.sample()
        
        loss = 10** -.5  *  criterion(logits, labels)
        model.zero_grad()
        loss.backward(retain_graph=True)
        diag.update(batch_size=images.size(0))
`

**Comment by hummat** [comment]
Hi @simomagi, yes, you could do this here. We opted to perform scaling after the estimation step and to treat the magnitude as a hyperparameter to be optimized for lowest calibration error instead of using the analytically correct scaling factor. This is often necessary as the Gaussian approximation is too inaccurate and needs to be regularized.
